docs/class5/connexml.md

```diff
@@ -319,23 +319,34 @@ An existing call is transferred to a different destination using the `Dial` ver
     </Response>
     ```
 
-|**Attribute**|**Description**|**Options**|**Default Method**
+|**Attribute**|**Description**|**Options**|**Default Method**|
 |-------------|---------------|-----------|-------|
 |`callerID`|Caller ID that must be a valid E.164 format number|
 |`fromDisplayName`|The fromDisplayName string to be used as the caller id name (SIP From Display Name) presented to the destination. The string should have a maximum of 128 characters, containing only letters, numbers, spaces, and -_~!.+ special characters. If omitted, the display name will be the same as the number in the callerId field|
 |`hangupOnStar`|By tapping the `*` key on their phone, the initial caller can hang up on the called party using the hangupOnStar attribute. It doesn't apply for `Conference` noun|`true`, `false`| `false`|
 |`ringTone`|The ringback tone played back to the caller|`at`,`au`,`bg`,`br`,<br>`be`,`ch`,`cl`,`cn`,`cz`,</br>`de`,`dk`,`ee`,`es`,`fi`,<br>`fr`,`gr`,`hu`,`il`,`in`,<br>`it`,`lt`,`jp`,`mx`,`my`,<br>`nl`,`no`,`nz`,`ph`,`pl`,<br>`pt`,`ru`,`se`,`sg`,<br>`th`,`uk`,`us`,`us-old`,`tw`,<br>`ve`,`za`|`us`|
 |`timeout`|timeout in <Dial> lets you specify the maximum waiting time in seconds for the dialed party to answer|||
 
-|**Noun**|**Description**|
-|-------------|----------|
+|**Noun**|**Description**|**Default Method**|
+|--------|---------------|------------------|
 |`Number`|Its is an E.164 phone number|
 |`Queue`|Its a queue name|
 |`Client`|It specifies a client identifier to dial|
 |`Conference`|You can connect to a conference room using the `Dial` verb's `Conference` noun|
+|`Voicemail`| Your voicemails can be played using the `Voicemail` noun. [Click here](https://docs.connexcs.com/class5/voicemail/#voicemail-dialpad-options) to know more options for the Voicemail Dialpad options|
+|`barge`|Allows you to join an ongoing call, without alerting the participants| `whisper`, `bridge`, `commands`|
+
+|**Noun**|**Attribute**|**Description**|**Options**|**Default Method**|
+|--------|--------|-----|-----------|------------------|
+|`Voicemail`|`Voicemail Inbox`| It sends you to the inbox to leave a message|
+|`Barge`| `whisper`|if enabled allows you to speak with the 3rd-party|`a`(aleg), `b`(bleg), `ab`(both legs)|`ab`|
+||`bridge`|listening in on a conference call where the eavesdropper isn't a participant but can still monitor the conversation|`a`(aleg), `b`(bleg),`ab` (both legs)|`ab`|
+||`command`|DTMF signals during eavesdrop|`true`, `false`|`true`|
 
 !!! Info
-    `Conference` is similar to how the `Number` noun lets you connect to another phone number.
+    1. `Conference` is similar to how the `Number` noun lets you connect to another phone number.
+    2. DTMF signals/ Commands in `Barge`:
+    `2 to speak with the uuid` <br> `1 to speak with the other half` <br> `3 to engage a three way` <br> `0 to restore eavesdrop to next channel`
```

Comment:
2: Speak with the A leg.
1: Speak with the B leg.
3: Engage in a three-way call.
0: Restore eavesdrop mode.
*: Move to the next channel.

Reply:
done